src/components/react_canvas/bar_geometries.tsx

```diff
@@ -58,49 +55,46 @@ export class BarGeometries extends React.PureComponent<BarGeometriesDataProps, B
         bar.geometryId,
         this.props.highlightedLegendItem,
         sharedStyle,
-        customOpacity,
+        seriesStyle.rect.opacity,
         individualHighlight,
       );
+      const key = `bar-${index}`;
 
       if (this.props.animated) {
         return (
           <Group key={index}>
             <Spring native from={{ y: y + height, height: 0 }} to={{ y, height }}>
               {(props: { y: number; height: number }) => {
-                const barProps = buildBarProps({
-                  index,
+                const barProps = buildBarRenderProps(
                   x,
-                  y: props.y,
+                  props.y,
                   width,
-                  height: props.height,
-                  fill: color,
-                  stroke: border.stroke,
-                  strokeWidth: border.strokeWidth,
-                  borderEnabled: border.visible,
+                  props.height,
+                  color,
+                  seriesStyle.rect,
+                  seriesStyle.rectBorder,
                   geometryStyle,
-                });
+                );
 
-                return <animated.Rect {...barProps} />;
+                return <animated.Rect key={key} {...barProps} />;
               }}
             </Spring>
           </Group>
         );
       } else {
-        const barProps = buildBarProps({
-          index,
+        const barProps = buildBarRenderProps(
           x,
           y,
           width,
           height,
-          fill: color,
-          stroke: border.stroke,
-          strokeWidth: border.strokeWidth,
-          borderEnabled: border.visible,
+          color,
+          seriesStyle.rect,
+          seriesStyle.rectBorder,
           geometryStyle,
-        });
+        );
         return (
           <React.Fragment key={index}>
-            <Rect {...barProps} />
+            <Rect key={key} {...barProps} />
```

Comment:
Do you want to allow `barProps.key` to potentially override the `key` prop?
```suggestion
            <Rect {...barProps} key={key}/>
```

Reply:
nice catch!

Reply:
solved in 111a465 (with a check on all the keys props)